internal/services/manageddevopspools/managed_devops_pool_resource.go

```diff
@@ -0,0 +1,242 @@
+package manageddevopspools
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/identity"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/location"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/devopsinfrastructure/2025-01-21/pools"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+var _ sdk.Resource = ManagedDevOpsPoolResource{}
+
+type ManagedDevOpsPoolResource struct{}
+
+func (ManagedDevOpsPoolResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.StringMatch(
+				regexp.MustCompile(`^[a-zA-Z0-9][a-zA-Z0-9-.]*[a-zA-Z0-9-]$`),
+				"Name must consist of alphanumeric characters, '.' or '-'. Name must also start with alphanumeric characters and cannot end with '.'.",
+			),
+		},
+		"resource_group_name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+		},
+		"location":      commonschema.Location(),
+		"agent_profile": AgentProfileSchema(),
+		"dev_center_project_resource_id": {
```

Comment:
could you reuse `dev_center_project_id` in `dev_center_project_pool_resource.go` ?

Reply:
Good suggestion. I have updated., please check